tests/cpp/test_gpu_view.cpp

```diff
@@ -368,33 +367,54 @@ std::vector<ReshapeExample> reshape_after_reduce_examples = {
     {{1, 27454, 1, 2}, {1, 3922, 1, 7}},
     {{1, 7844, 1, 7}, {1, 1961, 4}}};
 
-namespace {
-using ReshapeBeforeReduction = NVFuserFixtureParamTest<ReshapeExample>;
-TEST_P(ReshapeBeforeReduction, FusionReshapeBeforeReduction) {
-  const auto& [input_shape, output_shape] = GetParam();
-  maybeClearAllocator(); // Shmoo tests can occupy a lot of memory
-  reductionViewAddFusion(
-      input_shape, output_shape, /*reshape_before_reduction=*/true);
+namespace ReshapeReduction {
+
+struct ReshapeReductionParam {
+  ReshapeExample reshape_example;
+  bool has_implicit_broadcast;
+  bool reshape_before_reduction;
+};
+
+std::vector<ReshapeReductionParam> generateReshapeReductionParams() {
+  // For each reshape, test with and without implicit broadcast
+  int total_tests =
+      2 * (all_reshape_examples.size() + reshape_after_reduce_examples.size());
+  std::vector<ReshapeReductionParam> params;
+  params.reserve(total_tests);
+  for (auto reshape_before_reduction : {true, false}) {
+    const auto& examples = reshape_before_reduction
+        ? all_reshape_examples
+        : reshape_after_reduce_examples;
+    for (auto has_implicit_broadcast : {false, true}) {
```

Comment:
There's a thing called testing::Combine, which I'd probably use and simplify: https://google.github.io/googletest/reference/testing.html#INSTANTIATE_TEST_SUITE_P

Reply:
Seems a way to reduce lines of code. This test is not using `all combinations`, e.g. shape of input & output depend on whether reshape is before or after reduction. Will prioritize using this func when the test is a Cartesian product of all paras.

Reply:
Got it! 

Nit: `has_implicit_broadcast` seems to be orthogonal, so could still benefit. 

Another nit: Instead of 
```
for (cond : {false, true}) {
  examples = cond ? a_examples : b_examples;
  for (example : examples) {
    params.push_back(...);
  }
}
```
I'd flatten it slightly so the code is more direct. 
```
for (example : a_examples) {
  params.push_back(..., /*cond=*/true);
}
for (example : b_examples) {
  params.push_back(..., /*cond=*/false);
}
```

(I'm not requesting changes. Just some nits to keep in mind for future PRs.)